gettsim/pre_processing/policy_for_date.py

```diff
@@ -49,61 +50,146 @@
+                            )
+                    tax_data[param] = get_piecewise_parameters(
+                        tax_data[param],
+                        param,
+                        func_type=tax_data[param]["type"].split("_")[1],
+                    )
+            for key in ["type", "progressionsfaktor"]:
+                tax_data[param].pop(key, None)
+
+    return tax_data
+
+
+def check_groups(groups):
+    """Check group argument for correct format and transfer to list.
+
+    Parameters
+    ----------
+    groups : list, str
+            The group or a list of groups which parameters are loaded. Default is
+             all parameters
+
+    Returns
+    -------
+    List of groups to be loaded.
+
+    """
 
-    all_params_groups = [
-        "eink_st",
-        "eink_st_abzuege",
-        "soli_st",
-        "arbeitsl_geld_2",
-        "arbeitsl_geld",
-        "soz_vers_beitr",
-        "unterhalt",
-        "abgelt_st",
-        "wohngeld",
-        "kinderzuschlag",
-        "kindergeld",
-        "elterngeld",
-    ]
     if type(groups) == list:
-        group_list = groups
+        return groups
     elif type(groups) == str:
         if groups == "all":
-            group_list = all_params_groups
-        elif groups in all_params_groups:
-            group_list = [groups]
+            return INTERNAL_PARAM_GROUPS
+        elif groups in INTERNAL_PARAM_GROUPS:
+            return [groups]
         else:
             raise ValueError(f"{groups} is not a category for groups.")
     else:
         raise ValueError(f"{groups} is not a string or list.")
 
-    params_dict = {}
-    year = policy_date.year
 
-    for group in group_list:
-        tax_data = load_data(policy_date, group)
+def check_date(policy_date):
+    """Check the policy date for different input formats.
 
-        for param in tax_data:
-            if type(tax_data[param]) == dict:
-                if "type" in tax_data[param]:
-                    if tax_data[param]["type"].startswith("piecewise"):
-                        if "progressionsfaktor" in tax_data[param]:
-                            if tax_data[param]["progressionsfaktor"]:
-                                tax_data[param] = add_progressionsfaktor(
-                                    tax_data[param], param
-                                )
-                        tax_data[param] = get_piecewise_parameters(
-                            tax_data[param],
-                            param,
-                            func_type=tax_data[param]["type"].split("_")[1],
-                        )
-                for key in ["type", "progressionsfaktor"]:
-                    tax_data[param].pop(key, None)
+    Parameters
+    ----------
+    policy_date : datetime.date, str, int
+                The date for which the policy system is set up.
 
-        tax_data["datum"] = policy_date
-        params_dict[group] = tax_data
+    Returns
+    -------
+    policy_date : datetime.date
```

Comment:
Maybe I would advocate for `pd.Timestamp` instead of `datetime.date`. It is very similar, but we use pandas throughout gettsim. Using datetime.date, we have an additional "foreign" dtype. But feel free to disregard it. No need for a discussion.

Reply:
Using datetime.date everywhere!